app/code/Magento/Inventory/Ui/Component/Control/Stock/DeleteButton.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Ui\Component\Control\Stock;
+
+use Magento\Framework\View\Element\UiComponent\Control\ButtonProviderInterface;
+use Magento\InventoryApi\Api\Data\StockInterface;
+
+/**
+ * Class DeleteButton
+ */
+class DeleteButton extends GenericButton implements ButtonProviderInterface
```

Comment:
it's preferable to use Composition over Inheritance,
so try to not inherit from GenericButton, but just inject it as Constructor Dependency